src/TensorNetwork.jl

```diff
@@ -45,6 +45,29 @@ function TensorNetwork{A}(tensors; meta...) where {A}
     return tn
 end
 
+function Base.replace!(tn::TensorNetwork, old_tensor::Tensor, new_tensor::Tensor)
```

Comment:
Use better a `Pair` (i.e. `old_tensor => new_tensor`) because it makes more sense visually and we can put many `Pair`s (a vararg).
```suggestion
function Base.replace!(tn::TensorNetwork, mapping::Pair{Tensor, Tensor})
```